src/main/java/delight/nashornsandbox/internal/LinkedHashMapSecuredJsCache.java

```diff
@@ -0,0 +1,40 @@
+package delight.nashornsandbox.internal;
+
+import java.util.LinkedHashMap;
+import java.util.function.Supplier;
+
+import delight.nashornsandbox.SecuredJsCache;
+
+/**
+ * Default implementation of {@link SecuredJsCache}, uses a {@link LinkedHashMap}
+ * as its cache and is not thread-safe. Also, mixing scripts with missing braces
+ * allowed and not allowed is forbidden.
+ */
+class LinkedHashMapSecuredJsCache implements SecuredJsCache {
+
+	private final LinkedHashMap<String, String> map;
+	private final boolean allowNoBraces;
+
+	public LinkedHashMapSecuredJsCache(LinkedHashMap<String, String> linkedHashMap, boolean allowNoBraces) {
+		this.map = linkedHashMap;
+		this.allowNoBraces = allowNoBraces;
+	}
+
+	@Override
+	public String getOrCreate(String js, boolean allowNoBraces, Supplier<String> producer) {
+		assertConfiguration(allowNoBraces);
+		String result = map.get(js);
```

Comment:
Is this thread-safe here? Maybe it needs to be synchronized so that get() and put() are always executed together?

Reply:
This class is supposed to be a reflection of the status quo, and as the javadoc says, is not intended to be thread-safe. 

Why?
I'm working with the assumption that the sandbox itself is not thread-safe, and because currently, the cache is per-sandbox, the new default cache, which would also be per-sandbox, does not need to be thread-safe either. As this class is package private, it should not be possible for the user to re-use instances of it across different sandbox instances.

Maybe I am wrong about that initial assumption?

Also, I'd like to state again, that this pull request contains no actual implementation of a concurrent cache, only the interface to have the user put one in.